app/models/package_manager/base.rb

```diff
@@ -217,26 +205,25 @@ def self.new_names
 
     def self.save_dependencies(mapped_project)
       name = mapped_project[:name]
-      proj = Project.find_by(name: name, platform: db_platform)
-      proj.versions.includes(:dependencies).each do |version|
-        next if version.dependencies.any?
+      db_project = Project.find_by(name: name, platform: db_platform)
+      db_project.versions.includes(:dependencies).each do |db_version|
+        next if db_version.dependencies.any?
 
         deps = begin
-                 dependencies(name, version.number, mapped_project)
+                 dependencies(name, db_version.number, mapped_project)
                rescue StandardError
                  []
                end
-        next unless deps&.any? && version.dependencies.empty?
 
         deps.each do |dep|
-          next if dep[:project_name].blank? || version.dependencies.find_by_project_name(dep[:project_name])
+          next if dep[:project_name].blank? || db_version.dependencies.any? { |d| d.project_name == dep[:project_name] }
```

Comment:
Here's the db win in this PR: we've avoided an N+1 above with `db_project.versions.includes(:dependencies)`, but then reintroduced the N+1 and made it N^2+1 ( by doing `version.dependencies.find_by_project_name` for each dep for each version 😆 . This should fix.

Reply:
Now *that* is what I'm talking about! 💪